nixos/modules/services/security/oauth2_proxy_nginx.nix

```diff
@@ -25,10 +25,41 @@ in
     };
 
     virtualHosts = mkOption {
-      type = types.listOf types.str;
-      default = [];
+      type = let
+        vhostSubmodule = types.submodule {
+          options = {
+            allowed_groups = mkOption {
+              type = types.nullOr (types.listOf types.str);
```

Comment:
Should we assert that people do not set strings like &? or so here?

Reply:
I'm actually not too sure, maybe we can urlencode this? I'll test

Reply:
Looks like urlencoding does work.

Reply:
nice, that's probably even better